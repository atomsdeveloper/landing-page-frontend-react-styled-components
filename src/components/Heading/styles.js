import styled, { css } from 'styled-components';

const titleSize = {
    default: (theme) => css`
        font-size: ${theme.fontSize.default};
    `,
    small: (theme) => css`
        font-size: ${theme.fontSize.small};
    `,
    medium: (theme) => css`
        font-size: ${theme.fontSize.medium};
    `,
    big: (theme) => css`
        font-size: ${theme.fontSize.big};
    `,
};
const titleCase = (upperCase) => css`
    text-transform: ${upperCase ? "uppercase" : "none"};
`;
const Container = styled.h1`
    ${({theme, light, size, upperCase}) => css`
        color: ${light ? theme.colors.textColorDark : theme.colors.textColorLight};
        ${ titleSize[size](theme) };
        ${ titleCase(upperCase) }
    `}
`;
export default Container;