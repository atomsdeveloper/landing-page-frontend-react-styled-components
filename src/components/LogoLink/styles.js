import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({theme, light}) => css` 
        display: flex;
        align-items: center;
        color: ${!light ? theme.colors.textColorDark : theme.colors.textColorLight};
        text-decoration: none;
        
        > img {
            max-height: 3rem;
        }
    `};
`;