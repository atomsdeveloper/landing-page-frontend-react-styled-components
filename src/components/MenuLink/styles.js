import styled, { css } from 'styled-components';

const Container = styled.a`
    ${({theme}) => css`
        display: block;
        text-decoration: none;
        font-size: ${theme.fontSize.default};
        padding: ${theme.fontSize.default};
        color: ${theme.colors.textColorDark};
        position: relative;

        &::after {
            content: '';
            position: absolute;
            bottom: : 0;
            left: 0;
            width: 0;
            height: 0.2rem;
            background: ${theme.colors.textColorLight};
        }

        &:hover::after {
            left: 25%;
            width: 50%;
        }
    `}

`;
export default Container;