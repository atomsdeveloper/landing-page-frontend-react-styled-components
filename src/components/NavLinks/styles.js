import styled, { css } from 'styled-components';

const Container = styled.nav`
    ${() => css`
        display: flex;
        flex-flow: row wrap;

        @media (max-width: 768px) {
          flex-flow: column wrap;
          align-content: center;
        }
  `}
`;
export default Container;