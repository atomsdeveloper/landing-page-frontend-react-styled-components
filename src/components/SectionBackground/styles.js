import styled, { css } from 'styled-components';

const containerBackgroundActive = (theme) => css`
    color: ${theme.colors.textColorLight};
    background-color: ${theme.colors.bgColorDark};

`;
export const Container = styled.div`
    ${({theme, bg}) => css`
        display: flex;
        align-items: center;
        min-height: 100vh;
        ${bg && containerBackgroundActive(theme)};
        color: ${theme.colors.textColorDark};
        background-color: ${theme.colors.bgColorLight};
    `};
`;