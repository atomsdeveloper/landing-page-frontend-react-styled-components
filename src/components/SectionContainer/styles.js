import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${() => css`
        width: 100%;
        margin: 0 auto;
        padding: 3.2rem;
        max-width: 120rem;
    `};
`;