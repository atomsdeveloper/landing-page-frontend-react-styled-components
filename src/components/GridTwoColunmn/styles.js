import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, bg }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: 2rem;
    @media ${theme.media.mobile} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: 3rem;
      color: ${bg ? theme.colors.textColorLight : theme.colors.textColorDark};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.mobile} {
      margin-bottom: 2rem;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css``}
`;

export const Image = styled.img`
  ${() => css``}
`;