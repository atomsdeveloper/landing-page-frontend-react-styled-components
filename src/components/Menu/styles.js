import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Container as Heading } from '../Heading/styles'

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    border-bottom: ${theme.colors.textColorDark};
    background: ${theme.colors.bgColorLight};
    transition: all 300ms ease-in-out;

    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.mobile} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${visible && menuVisible(theme)}

      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      & ${Heading} {
        padding-bottom: 1rem;
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
    ${({theme}) => css`
        color: ${theme.colors.textColorDark};
        display: flex;
        justify-content: space-between;
        align-items: center;

        @media (max-width: 768px) {
            display: block;
            text-align: center;
            padding: 3.2rem 0;
        }    
    `};
`;

export const Button = styled.div`
    ${({theme, visible}) => css`
       display: flex;
       align-items: center;
       justify-content: center;
       z-index: 6;
       position: fixed;
       top: 2rem;
       right: 2rem;
       width: 4rem;
       height: 4rem;
       background: ${theme.colors.bgColorDark};
       color: ${theme.colors.textColorLight};
       border: none;
       pointer-events: ${visible ? 'none' : 'all'};
    `}; 

    @media (max-width: 768px) {
        display: flex;
        align-items: center;
        justify-content: center;
    }

    > svg {
        width: 2.5rem;
        height: 2.5rem
    }
`;