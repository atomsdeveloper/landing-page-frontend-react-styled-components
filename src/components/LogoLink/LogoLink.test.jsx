import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { LogoLink } from '.';

describe('<LogoLink />', () => {
    it('should render text logo', () => {
        renderTheme(<LogoLink link="#target" text='Olá mundo'/>);
        expect(screen.getByRole('heading', {name: 'Olá mundo'}).firstChild).toHaveAttribute('href', '#target');
    });

    it('should render image logo', () => {
        renderTheme(<LogoLink link="#target" text='Olá mundo' src='image.jpg'/>);
        expect(screen.getByAltText('Olá mundo').toHaveAttribute('scr', 'image.jpg'));
    });

    it('should match snapshot', () => {
        const { container } = renderTheme(<LogoLink link="#target" text='Olá mundo' src='image.jpg'/>);
        expect(container.firstChild).toMatchSnapshot();
    });
})