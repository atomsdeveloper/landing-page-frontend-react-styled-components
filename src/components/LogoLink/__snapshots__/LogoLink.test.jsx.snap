// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
.c1 {
  display: flex;
  align-items: center;
  color: #ffff;
  text-decoration: none;
}

.c1 >img {
  max-height: 3rem;
}

.c0 {
  color: #ffff;
  font-size: 1.6rem;
  text-transform: uppercase;
}

<h1
  class="c0"
>
   
  <a
    class="c1"
    href="#target"
  >
    Ol√° mundo
  </a>
   
</h1>
`;
